sdk/formrecognizer/Azure.AI.FormRecognizer/src/autorest.md

```diff
@@ -35,6 +33,28 @@ directive:
     });
 ```
 
+## Seal single value enums
```

Comment:
Adding these directives to ensure we'll have no code changes in Form Recognizer. The generated code will end up being the same.

Reply:
I'm not gonna lie ... I have no idea what these are doing or what the intent is.   Can we please add a short explanation as part of this section for future readers?

Reply:
I'll add an explanation as suggested, but just to give you context:

This is the enum we're modifying in the original swagger ([link here](https://github.com/Azure/azure-rest-api-specs/blob/7043b48f4be1fdd40757b9ef372b65f054daf48f/specification/cognitiveservices/data-plane/FormRecognizer/stable/v2.1/FormRecognizer.json#L981-L990)):
```json
{
  "name": "op",
  "in": "query",
  "description": "Specify whether to return summary or full list of models.",
  "required": true,
  "type": "string",
  "enum": [
    "full"
  ]
}
```

By default, CodeGen will create a whole extensible enum readonly struct to represent it in our code, and call it `Enum6`. Some methods in the internal auto-generated `RestClient` will also require an `Enum6` object to be specified, even though we'll always be passing the same value (`Enum6.Full`):
```cs
// Auto-generated method
FormRecognizerRestClient.ListCustomModels(Enum6 op, CancellationToken cancellationToken = default);

// We will always call it like this
restClient.ListCustomModels(Enum6.Full, cancellationToken);
```

The `modelAsString` flag tells CodeGen whether we should generate an extensible enum or not (`true` for an extensible enum, `false` for a constant). When we make it `false`, single-value enums will become regular strings, and since they can only assume 1 possible value, the generated code will not even ask for us to provide it. The code above becomes:
```cs
// Auto-generated method
FormRecognizerRestClient.ListCustomModels(CancellationToken cancellationToken = default);

// We will always call it like this
restClient.ListCustomModels(cancellationToken);
```

We previously had the `seal-single-value-enum-by-default` flag enabled for us. It set `modelAsString` to `false` for every single-value enum in the swagger. However, it's going to be deprecated, so we need to include them manually now.